packages/access-client/src/agent-data.js

```diff
@@ -171,12 +171,18 @@ export const isSessionProof = (delegation) =>
  * Get a map from CIDs to the session proofs that reference them
  *
  * @param {AgentData} data
+ * @param {Ucanto.DID} [issuer] - if provided, will only return session proofs issued by this id
  * @returns {Record<string, Ucanto.Delegation>}
  */
-export function getSessionProofs(data) {
+export function getSessionProofs(data, issuer) {
   /** @type {Record<string, Ucanto.Delegation>} */
   const proofs = {}
   for (const { delegation } of data.delegations.values()) {
+    if (issuer !== undefined && issuer !== delegation.issuer.did()) {
+      // delegation doesn't match issuer
+      // eslint-disable-next-line no-continue
```

Comment:
add to config in package.json?

Reply:
good idea! did it in commit 7248d915fa53d7389f09ae4504b39122bf589dee